### ISSUE TITLE
Storage.getUrl("ExampleKey") should be able to throw a meaningful exception if the file does not exist in the storage

### ISSUE THREAD

**Issue by kvramyasri7** [aws-amplify/amplify-android#2404]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Not applicable

### Amplify Categories

Storage

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

When we call getUrl from storage category we are receiving a presigned url even though the file does not exist.The expected behavior is to return a meaningful exception if the file does not exist in the storage.

### Reproduction steps (if applicable)

amplify add storage
Call Amplify.Storage.getUrl("NonexistedFile");. call getUrl with a non-existed File.

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by kvramyasri7** [comment]
Hello All, Wondering when this is going to be resolved. 😉. Just an ETA will be fine or you can circle back to me when this is fixed.

**Comment by phantumcode** [comment]
Fix released in version[ 2.20.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.20.0)
